config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -34,11 +37,152 @@
         this.bundleDocumentBuilder = bundleDocumentBuilder;
     }
 
-    public Element build(Bundle bundle, EntityBuilder.BundleType bundleType, Document document) {
+    public Map<String, Element> build(Bundle bundle, EntityBuilder.BundleType bundleType, Document document, String bundleName, String bundleVersion) {
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
 
-        return bundleDocumentBuilder.build(document, entities);
+        List<AnnotatedEntity> annotatedEntities = new ArrayList<>();
+        // Filter the bundle to export only annotated entities
+        // TODO : Enhance this logic to support services and policies
+        final Map<String, Encass> encassEntities = bundle.getEntities(Encass.class);
+        encassEntities.entrySet().parallelStream().forEach(encassEntry -> {
+            Encass encass = encassEntry.getValue();
+            if (encass.getAnnotations() != null) {
+                encass.getAnnotations().forEach(annotation -> {
+                    AnnotatedEntity annotatedEntity = new AnnotatedEntity();
+                    if (annotation instanceof Map) {
+                        if (((Map) annotation).containsKey("type")) {
+                            try {
+                                switch (((Map) annotation).get("type").toString()) {
+                                    case ANNOTATION_TYPE_BUNDLE:
```

Comment:
Define a class for each annotation type and encapsulate all the annotation details inside the class.
Then apply annotation on AnnotatedEntity for each annotation
Like AnnotatedEntity.apply(Annotation)

This way we can get rid of all these if conditions

Reply:
I have implemented JsonDeserializer in my PR #212 to create the `Set<Annotaion>` which handles all these conditions.